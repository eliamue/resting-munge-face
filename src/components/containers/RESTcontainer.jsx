import React, { Component } from 'react';

export default class RESTContainer extends Component {
  state = {};

  handleChange = (event) => {
    this.setState(event.target.value);
  };

  handleSubmit = (event) => {
    this.setState(event.target.value);
  };

  render() {
    return <div></div>;
  }
}
