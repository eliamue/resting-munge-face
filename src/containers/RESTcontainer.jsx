import React, { Component } from 'react';
import Controls from '../components/resty/Controls';
import Header from '../components/resty/Header';

export default class RESTContainer extends Component {
  state = {
    url: '',
    route: '',
    body: '',
  };

  componentDidMount() {
    const pastRequests = JSON.parse(localStorage.getItem('storage'));
    if(pastRequests) {
      this.setState({ storage: pastRequests });
    }
  }

  handleChange = (event) => {
    this.setState({ [event.name]: event.value });
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    const { url, route, body } = this.state;
    if(route === 'GET') {
      await fetch(`${url}`, {
        route
      });
    } else {
      await fetch(`${url}`, {
        route,
        body: JSON.stringify(body)
      });
    }
  };

  render() {
    const { url, body, onChange, onSubmit } = this.state;
    return (
      <div>
        <Header />
        <Controls 
          url={url} 
          body={body} 
          onChange={onChange}
          onSubmit={onSubmit}
        />
      </div>
    );
  }
}
