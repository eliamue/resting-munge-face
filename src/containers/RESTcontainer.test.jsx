import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import RESTcontainer from './RESTcontainer';
import userEvent from '@testing-library/user-event';

describe('RESTcontainer', () => {
  it('uses method request and returns json response', () => {
    render(<RESTcontainer />);

    const text = screen.getByRole('textbox', { name: 'url-input' });
    expect(text).toMatchSnapshot();

    const urlInput = screen.findByPlaceholderText('https://');
    userEvent.type(
      urlInput,
      'https://botw-compendium.herokuapp.com/api/v2/category/monsters'
    );

    const submit = screen.findByRole('button');
    userEvent.click(submit);
  });
});
