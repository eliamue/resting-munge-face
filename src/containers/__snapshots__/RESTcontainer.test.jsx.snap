// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESTcontainer uses method request and returns json response 1`] = `
<input
  aria-label="url-input"
  id="url"
  name="url"
  placeholder="https://"
  type="url"
  value=""
/>
`;
